### PR TITLE
Add ability to use test config file to list tests to skip

### PR DESCRIPTION
**Pull request by dsplaisted**
Add support for passing a `-testConfigFile` parameter when running tests.  This allows individual tests to be skipped without having to modify the test source code.  This will be helpful to allow tests to be run on other repos, such as with https://github.com/dotnet/core-sdk/pull/1454.

The test config file looks like this

```xml
<Tests>
  <Method Name="Microsoft.NET.Clean.Tests.GivenThatWeWantToCleanAHelloWorldProject.It_can_clean_and_build_without_using_rebuild"
          Skip="true"
          Issue="https://github.com/dotnet/sdk/issues/1234"
          Reason="Reason for skip"/>
</Tests>
```

The `Issue` and `Reason` attributes aren't currently used, they are merely informational to help us keep such a list up-to-date.

### CODE REVIEW

**Comment by wli3** [comment]
tests?

File: src/Tests/Microsoft.NET.TestFramework/TestCommandLine.cs [hunk 1]
```diff
@@ -102,6 +109,32 @@ public static TestCommandLine Parse(string[] args)
             return ret;
         }
 
+        public static List<string> GetXunitArgsFromTestConfig(string testConfigFile)
```
Comment (wli3) [inline src/Tests/Microsoft.NET.TestFramework/TestCommandLine.cs]: dotnet test is not xunit specific tho

**Comment by wli3** [comment]
is test config a standard format? could you give me the link to the schema?

**Comment by dsplaisted** [comment]
@wli3 This is just a format I came up with.  It's not part of the product, it lets us specify which of our tests should be skipped, and all of our tests are Xunit.  It doesn't apply to `dotnet test`, it applies when you "run" the test project, or the dotnet tool created from it.

**Comment by wli3** [comment]
I see, it is only in tests, not the product

**Review by wli3 — APPROVED** [review]

**Pull request merged**